### PR TITLE
10 Create Team API Endpoint

### LINKED ISSUE #10: Create team endpoint
## Basic Information
Create a new API endpoint for the "Team" db table using the Django REST framework. This task is split into multiple small parts that each team member can tackle! It should be under the folder server/api/team

1. Create Team Model in `models.py`
2. Create Views in `views.py`
It should have views for the following operations:
- create a new team
- read/get details of team
- update details of the team
- Delete team, be careful with this we dont want to lose existing data if the team id is a FK in another table
3.  Add url patterns to `urls.py`
4.  Register the app for the admin

You will be required to research how to complete this task!

TIP: You can use postman to test your endpoints!

### PR DESCRIPTION
**Pull request by YunnTun**
## Change Summary
**[Briefly summarise the changes that you made. Just high-level stuff]**

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [ ] The pull request title has an issue number
- [ ] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**
Resolve #10

### CODE REVIEW

File: server/teams/models.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    user_id = models.AutoField(primary_key=True)
+    username = models.CharField(unique=True)
+    first_name = models.CharField(max_length = 25)    
+    last_name = models.CharField(max_length = 25)    
+    email = models.EmailField(max_length = 254, unique=True)
+    team_id = models.ForeignKey(Team, null=True, blank=True, on_delete=models.SET_NULL, related_name='memberOf')
+    is_admin = models.BooleanField()
+
+    def __str__(self):
+        return f"{self.username}"
```
Comment (ChantelleKerr) [inline server/teams/models.py:28]: This shouldnt be in here. Another issue has been made for the User

File: server/teams/templates/create_team.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+<form action="{% url 'teams:create_team' %}" method="POST">
+    {% csrf_token %}
+    {{ form.as_p }}
+
+    <input type="submit">
+</form>
+{% endif %}
+
+{% if err %}
+{{ err }}
```
Comment (ChantelleKerr) [inline server/teams/templates/create_team.html:20]: We wont be using templates, we are using vue for the frontend

File: server/api/settings.py [hunk 3]
```diff
@@ -40,6 +40,7 @@
     "django.contrib.sessions",
     "django.contrib.messages",
     "django.contrib.staticfiles",
+    "teams"
```
Comment (ChantelleKerr) [inline server/api/settings.py:43]: the teams folder should be under api -> api.team.

the folder should also be "team" not "teams"

File: server/api/team/tests.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# from django.test import TestCase
+
+# Create your tests here.
```
Comment (frinzekt) [inline server/api/team/tests.py:3]: please learn how to do testing see 
here's an example https://www.django-rest-framework.org/api-guide/testing/#api-test-cases

Comment (ChantelleKerr) [reply server/api/team/tests.py:3]: Dont worry about testing - we can create a new issues for this :)

File: server/api/team/views.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+def get_teams(request):
+    teams = Team.objects.all()
+    serializer = TeamSerialiser(teams, many=True)
+    return Response(serializer.data)
+
+
+@api_view(['PUT'])
+def update_team(request, team_id):
+    team = Team.objects.get(team_id=team_id)
+    if team.is_public is False:
```
Comment (ChantelleKerr) [inline server/api/team/views.py:33]: what is is_public?

File: server/api/team/views.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            serializer.save()
+        return Response(serializer.data)
+    else:
+        return Response("Team is not private")
+
+
+@api_view(['DELETE'])
+def delete_team(request, team_id):
+    team = Team.objects.get(team_id=team_id)
+    if team.is_public is False:
```
Comment (ChantelleKerr) [inline server/api/team/views.py:45]: the team model has no is_public attribute

File: .vscode/settings.json [hunk 7]
```diff
@@ -1,3 +1,6 @@
 {
     "prettier.tabWidth": 2
+    
 }
+
+
```
Comment (ChantelleKerr) [inline .vscode/settings.json:6]: we have a bunch of empty lines here that are not needed

**Review by ChantelleKerr — CHANGES_REQUESTED** [review]
Nice job, some small changes, Thank you :)

File: server/api/team/tests.py (hunk 4 above)
Comment (frinzekt) [reply server/api/team/tests.py:3]: Then create a new issue Chantelle!

File: server/api/team/views.py (hunk 6 above)
Comment (YunnTun) [reply server/api/team/views.py:45]: Sorry, my bad. Too much copy/pasting hahah

**Review by ChantelleKerr — APPROVED** [review]
Great work! Approved.

**Pull request merged**